packages/ra-ui-materialui/src/list/List.spec.tsx

```diff
@@ -340,4 +343,182 @@ describe('<List />', () => {
+        });
+        it('should be displayed if an item is selected', async () => {
+            render(
+                <Default>
+                    <Child />
+                </Default>
+            );
+            await waitFor(() => {
+                expect(screen.queryAllByRole('checkbox')).toHaveLength(11);
+            });
+            fireEvent.click(screen.getAllByRole('checkbox')[1]);
+            expect(
+                await screen.findByRole('button', { name: 'Select all' })
+            ).toBeDefined();
+        });
+        it('should not be displayed if all item are manyally selected', async () => {
+            render(
+                <Default
+                    dataProvider={testDataProvider({
+                        getList: () =>
+                            Promise.resolve<any>({
+                                data: [
+                                    {
+                                        id: 0,
+                                        title: 'War and Peace',
+                                        author: 'Leo Tolstoy',
+                                        year: 1869,
+                                    },
+                                    {
+                                        id: 1,
+                                        title: 'Pride and Prejudice',
+                                        author: 'Jane Austen',
+                                        year: 1813,
+                                    },
+                                ],
+                                total: 2,
+                            }),
+                    })}
+                >
+                    <Child />
+                </Default>
+            );
+            await waitFor(() => {
+                expect(screen.queryAllByRole('checkbox')).toHaveLength(3);
+            });
+            fireEvent.click(screen.getAllByRole('checkbox')[0]);
+            await screen.findByText('2 items selected');
+            expect(
+                screen.queryByRole('button', { name: 'Select all' })
+            ).toBeNull();
+        });
+        it('should not be displayed if all item are selected with the "Select all" button', async () => {
+            render(
+                <Default>
+                    <Child />
+                </Default>
+            );
+            await waitFor(() => {
+                expect(screen.queryAllByRole('checkbox')).toHaveLength(11);
+            });
+            fireEvent.click(screen.getAllByRole('checkbox')[0]);
+            await screen.findByText('10 items selected');
+            fireEvent.click(screen.getByRole('button', { name: 'Select all' }));
+            await screen.findByText('13 items selected');
+            expect(
+                screen.queryByRole('button', { name: 'Select all' })
+            ).toBeNull();
+        });
+        it('should not be displayed if we reached de selectAllLimit by a manyally selection', async () => {
```

Comment:
```suggestion
        it('should not be displayed if we reached the selectAllLimit by a manual selection', async () => {
```
or `da` if you wanna play it gangsta style ^^